src/Mod/Part/App/WireJoiner.cpp

```diff
@@ -0,0 +1,2524 @@
+        }
+    }
+
+    void pushIntersection(std::set<IntersectInfo> &params, double param, const gp_Pnt &pt, const TopoDS_Shape &shape)
+    {
+        IntersectInfo info(param, pt, shape);
+        auto it = params.upper_bound(info);
+        if (it != params.end()) {
+            if (it->point.SquareDistance(pt) < myTol2) {
+                return;
+            }
+        }
+        if (it != params.begin()) {
+            auto itPrev = it;
+            --itPrev;
+            if (itPrev->point.SquareDistance(pt) < myTol2) {
+                return;
+            }
+        }
+        params.insert(it, info);
+        return;
+    }
+
+    struct SplitInfo {
+        TopoDS_Edge edge;
+        TopoDS_Shape intersectShape;
+        Box bbox;
+    };
+
+    // Try splitting any edges that intersects other edge
+    void splitEdges()
+    {
+        std::unordered_map<const EdgeInfo*, std::set<IntersectInfo>> intersects;
+
+        int i=0;
+        for (auto &info : edges)
+            info.iteration = ++i;
+
+        std::unique_ptr<Base::SequencerLauncher> seq(
+                new Base::SequencerLauncher("Splitting edges", edges.size()));
+
+        i = 0;
+        for (auto it=edges.begin(); it!=edges.end();++it) {
+            seq->next(true);
+            ++i;
+            auto &info = *it;
+            auto &params = intersects[&info];
+            checkSelfIntersection(info, params);
+
+            for (auto vit=boxMap.qbegin(bgi::intersects(info.box)); vit!=boxMap.qend(); ++vit) {
+                const auto &other = *(*vit);
+                if (other.iteration <= i) {
+                    // means the edge is before us, and we've already checked intersection
+                    continue;
+                }
+                checkIntersection(info, other, params, intersects[&other]);
+            }
+        }
+
+        i=0;
+        std::vector<SplitInfo> splitted;  // NOLINT
+        for (auto it=edges.begin(); it!=edges.end(); ) {
+            ++i;
+            auto iter = intersects.find(&(*it));
+            if (iter == intersects.end()) {
+                ++it;
+                continue;
+            }
+            auto &info = *it;
+            auto &params = iter->second;
+            if (params.empty()) {
+                ++it;
+                continue;
+            }
+
+            auto itParam = params.begin();
+            if (itParam->point.SquareDistance(info.p1) < myTol2)
+                params.erase(itParam);
+            params.emplace(info.firstParam, info.p1, TopoDS_Shape());
+            itParam = params.end();
+            --itParam;
+            if (itParam->point.SquareDistance(info.p2) < myTol2)
+                params.erase(itParam);
+            params.emplace(info.lastParam, info.p2, TopoDS_Shape());
+
+            if (params.size() <= 2) {
+                ++it;
+                continue;
+            }
+
+            splitted.clear();  // NOLINT
```

Comment:
What should I use here and for the next warnings? `NOLINTNEXTLINE`?

Reply:
No, rename the thing. "splitted" is not a word, and every spellchecker that is ever run on the codebase is going to flag it. `splitEdges` or something like that.

Reply:
I'd just rename the variable to what lint wants; it is better english.  Or maybe to "split_edge_info"

Reply:
I saw that `splitEdge` is already used in other methods and it's another type of variable, therefor to minimize confusion I simply renamed it `split`